### DISCUSSION TITLE
Multiple dashboards

### DISCUSSION THREAD

**Discussion by youzer-name** [jasonacox/Powerwall-Dashboard#119] [Show and tell]
Just a show & tell post to share how I currently have Grafana set up with multiple dashboards. I had already split my panels into a small Current Status dashboard and a larger one with everything else, but since I've added a bunch of new monthly panels I decided to split all the long-term charts off into their own dashboard for historical data.

My default dashboard is Current Status. It is designed to fit on my desktop or laptop screen without scrolling:

![image](https://user-images.githubusercontent.com/57046429/199770102-344cbf1e-5cd3-4953-a182-4f93dae0b115.png)

This defaults to the last 24 hours and automatically refreshes every 1 minute. I have the Tesla animation, the Energy Usage chart, total use and cost, plus current temp, current cloud cover, total battery capacity, and a panel showing the time of the latest data.

I've added panels at on the right side that are not related to the Powerwall. These show my HVAC, vitals of the Raspberry Pi that is running this project, and free memory stats on my home automation hubs, plus the status of my 2 Tesla cars (from TeslaMate). I also have panels for the current total Powerwall capacity and the time of the last data update.

Then I have a bigger dashboard called Powerwall-Monitor

This also defaults to the last 24 hours and refreshes every minute. The top shows a lot of the same information as the current status dashboard but there are some extra panels. At the bottom it has two additional rows that can expand or collapse: Vitals, and Weather

![image](https://user-images.githubusercontent.com/57046429/199731416-b65511fe-c548-4cfb-8bed-bc9df10ad94d.png)

The vitals section expands to show powerwall temps, frequency, votages, and capacity charts. I've converted all of these from the older chart types to the newer Time Series type. I enhanced the Frequency and Voltages charts to add a Grid Status indicator. A red line appears at the bottom of those charts when the grid status is "off grid". This image shows a brief off-grid period last week.

![image](https://user-images.githubusercontent.com/57046429/199736321-2d2294ca-bb7a-4e1a-8c2d-92786563de24.png)

Finally the Weather row expands to show the weather charts:

![image](https://user-images.githubusercontent.com/57046429/199736703-2be67def-0aa3-4e2b-b0d9-ae93625dc761.png)

I moved all the monthly and historical charts to a third dashboard called Powerwall-History. This one pulls from my MySQL database with Time of Use pricing. The default time period on this dashboard is 30 days and it doesn't automatically refresh.

At the top I have total kWh and costs for the time period, plus a grid status chart. Below that I have the daily charts that always show the last month, and my monthly charts that always show the last 12 months for home, solar, gird, powerwall, total cost, solar cost, grid cost, and effective price per kWh per month. Below that is the 'solar energy year' chart , the vitals charts, and the weather charts. I have the weather charts saved with only clouds and rain displayed and the other items can be shown by clicking on the legend to turn any or all of them on. Since the Energy Usage panel isn't on this dashboard and many of the charts aren't affected by the selected time period, it updates pretty quickly when switching time spans (for example, from last 30 days to last 1 year).

![image](https://user-images.githubusercontent.com/57046429/199755582-0fe6c4c9-3882-4777-88c2-fd5b70ae2fa3.png)

![image](https://user-images.githubusercontent.com/57046429/199803812-b893fb6e-ac35-4404-8533-3588ecd22637.png)

![image](https://user-images.githubusercontent.com/57046429/199803871-c2d15e88-5bda-4cf0-9807-97be336cfefc.png)

![image](https://user-images.githubusercontent.com/57046429/199756287-3b4551c2-4383-4645-8a5b-03f152d56e1c.png)

![image](https://user-images.githubusercontent.com/57046429/199756362-a983e167-d094-4367-be44-29b18ca53bf0.png)

**Comment by jasonacox** [comment]
@youzer-name Love this! You are brilliant. This gets back to my desire to create a library (marketplace) to store these different dashboards for others to use.  At the very minimum, you will inspire others to explore so thanks for posting this!!! 👍

**Comment by mcbirse** [comment]
Wow, these are amazing @youzer-name - I love what you have done with this!

**Comment by ViktorJp** [comment]
@youzer-name ... wow, that's awesome!  Would you be willing to share a copy of your dashboard by chance??  :)

**Reply by youzer-name** [reply]
@ViktorJp Sure, with the caveat that some parts of these dashboards aren't going to translate to other users' setups because I'm tapping into non-standard data sources. I exported them with the 'Export for sharing externally' option which I think helps with the import process. 

Here is the small 'current status' dashboard. Most of the panels on the right are not going to work as they are showing non-powerwall data.

[Current Status-1668701651261.json.txt](https://github.com/jasonacox/Powerwall-Dashboard/files/10033382/Current.Status-1668701651261.json.txt)

Here is the larger Powerwall Monitor. I don't have a Powerwall +, so I don't show the strings panel. The 'cost' panel on this dashboard hasn't been updated to use the TOU data yet.

[Powerwall-Monitor-1668701834082.json.txt](https://github.com/jasonacox/Powerwall-Dashboard/files/10033407/Powerwall-Monitor-1668701834082.json.txt)

Here is Powerwall-History. This one is very dependant on the MySQL copy of the data which is where it gets its monthly data and TOU costs.

[Powerwall-History-1668702082380.json.txt](https://github.com/jasonacox/Powerwall-Dashboard/files/10033437/Powerwall-History-1668702082380.json.txt)

**Reply by ViktorJp** [reply]
Thanks so much!  As soon as I get grafana 9 working, I'm going to be giving these a shot and see if I can customize by weeding out some of your custom pw sources!  :)  Again, great job on this!
